ui/imports/shared/popups/ImportCommunityPopup.qml

```diff
@@ -20,17 +20,26 @@ StatusDialog {
     title: qsTr("Import Community")
 
     signal joinCommunity(string communityId, var communityDetails)
+
     QtObject {
         id: d
         property string importErrorMessage
 
-        readonly property bool communityFound: (d.communityDetails !== null && !!d.communityDetails.name)
-        readonly property var communityDetails: {
+        property bool loading
+        property bool communityFound: (d.communityDetails !== null && !!d.communityDetails.name)
+        property var communityDetails: {
             if (isInputValid) {
+                loading = true
                 let key = isPublicKey ? Utils.getCompressedPk(publicKey) :
-                          root.store.getCommunityPublicKeyFromPrivateKey(inputKey);
-                return root.store.getCommunityDetails(key);
+                          root.store.getCommunityPublicKeyFromPrivateKey(inputKey, true /*importing*/);
+
+                const details = root.store.getCommunityDetails(key)
+                if (!!details) {
+                    loading = false
+                }
```

Comment:
```suggestion
                loading = !details
```

Reply:
Ah no, it's actually not the same thing in this case; we already set `loading = true` earlier but we want to set it to `false` only when `root.store.getCommunityDetails(key)` actually returned something